libafl/src/events/tcp.rs

```diff
@@ -0,0 +1,1364 @@
+{
+    TcpRestartingMgr::builder()
+        .shmem_provider(StdShMemProvider::new()?)
+        .monitor(Some(monitor))
+        .broker_port(broker_port)
+        .configuration(configuration)
+        .hooks(tuple_list!())
+        .build()
+        .launch()
+}
+
+/// Provides a `builder` which can be used to build a [`TcpRestartingMgr`], which is a combination of a
+/// `restarter` and `runner`, that can be used on systems both with and without `fork` support. The
+/// `restarter` will start a new process each time the child crashes or times out.
+#[cfg(feature = "std")]
+#[allow(clippy::default_trait_access, clippy::ignored_unit_patterns)]
+#[derive(TypedBuilder, Debug)]
+pub struct TcpRestartingMgr<EMH, MT, S, SP>
+where
+    S: UsesInput + DeserializeOwned,
+    SP: ShMemProvider + 'static,
+    MT: Monitor,
+    //CE: CustomEvent<I>,
+{
+    /// The shared memory provider to use for the broker or client spawned by the restarting
+    /// manager.
+    shmem_provider: SP,
+    /// The configuration
+    configuration: EventConfig,
+    /// The monitor to use
+    #[builder(default = None)]
+    monitor: Option<MT>,
+    /// The broker port to use
+    #[builder(default = 1337_u16)]
+    broker_port: u16,
+    /// The address to connect to
+    #[builder(default = None)]
+    remote_broker_addr: Option<SocketAddr>,
+    /// The type of manager to build
+    #[builder(default = TcpManagerKind::Any)]
+    kind: TcpManagerKind,
+    /// The amount of external clients that should have connected (not counting our own tcp client)
+    /// before this broker quits _after the last client exited_.
+    /// If `None`, the broker will never quit when the last client exits, but run forever.
+    ///
+    /// So, if this value is `Some(2)`, the broker will not exit after client 1 connected and disconnected,
+    /// but it will quit after client 2 connected and disconnected.
+    #[builder(default = None)]
+    exit_cleanly_after: Option<NonZeroUsize>,
+    /// Tell the manager to serialize or not the state on restart
+    #[builder(default = true)]
+    serialize_state: bool,
+    /// The hooks for `handle_in_client`
+    hooks: EMH,
+    #[builder(setter(skip), default = PhantomData)]
+    phantom_data: PhantomData<S>,
+}
+
+#[cfg(feature = "std")]
+#[allow(clippy::type_complexity, clippy::too_many_lines)]
+impl<EMH, MT, S, SP> TcpRestartingMgr<EMH, MT, S, SP>
+where
+    EMH: EventManagerHooksTuple<S> + Copy + Clone,
+    SP: ShMemProvider,
+    S: State + HasExecutions + HasMetadata,
+    MT: Monitor + Clone,
+{
+    /// Launch the restarting manager
+    pub fn launch(&mut self) -> Result<(Option<S>, TcpRestartingEventManager<EMH, S, SP>), Error> {
+        // We start ourself as child process to actually fuzz
+        let (staterestorer, _new_shmem_provider, core_id) = if env::var(_ENV_FUZZER_SENDER).is_err()
+        {
+            let broker_things = |mut broker: TcpEventBroker<S::Input, MT>, _remote_broker_addr| {
```

Comment:
this is not really working for remote host...
this is the only place that remote_broker_addr is every passed, and it completely ignores the address

Reply:
Yes would be good to fix it at some point